server/rollback_runner.go

```diff
@@ -0,0 +1,159 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common/log"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/bytebase/bytebase/plugin/db/mysql"
+)
+
+var (
+	generateRollbackSQLChan = make(chan *api.Task, 100)
+)
+
+// NewRollbackRunner creates a new rollback runner.
+func NewRollbackRunner(server *Server) *RollbackRunner {
+	return &RollbackRunner{
+		server: server,
+	}
+}
+
+// RollbackRunner is the rollback runner generating rollback SQL statements.
+type RollbackRunner struct {
+	server *Server
+}
+
+// Run starts the rollback runner.
+func (r *RollbackRunner) Run(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	r.retryGetRollbackSQL(ctx)
+	for {
+		select {
+		case task := <-generateRollbackSQLChan:
+			r.getRollbackSQL(ctx, task)
+		case <-ctx.Done(): // if cancel() execute
+			return
+		}
+	}
+}
+
+// retryGetRollbackSQL retries generating rollback SQL for tasks.
+// It is currently called when Bytebase server starts and only rerun unfinished generation.
+func (r *RollbackRunner) retryGetRollbackSQL(ctx context.Context) {
+	find := &api.TaskFind{
```

Comment:
We should also look at MySQL only. Also, is it guaranteed to get rollback SQL?

Reply:
> We should look at MySQL only here

It's hard to write in SQL, we filter MySQL in the loop below.

> Also, is it guaranteed to get rollback SQL?

As long as there's data change in the task.

Reply:
Then are we scanning a lot of tasks?

Reply:
Improved.